### PR TITLE
For #4507 Enable codecov on PRs 

### PR DESCRIPTION
**Pull request by colintheshots**
I'm testing for a bit. Don't land this until I remove the label.

### CODE REVIEW

File: automation/taskcluster/lib/tasks.py [hunk 1]
```diff
@@ -70,14 +70,17 @@ def craft_assemble_release_task(self, architectures, build_type, is_staging, ver
         )
 
         capitalized_build_type = upper_case_first_letter(build_type)
+        coverage_flag = ('-Pcoverage ' if run_coverage else '',)
```
Comment (mitchhentges) [inline automation/taskcluster/lib/tasks.py]: Does this make sense to be a tuple? Perhaps `coverage_flag = '-Pcoverage' if run_coverage else ''` would be cleaner?

File: automation/taskcluster/lib/tasks.py [hunk 2]
```diff
@@ -41,7 +41,7 @@ def __init__(
         self.date = arrow.get(date_string)
         self.trust_level = trust_level
 
-    def craft_assemble_release_task(self, architectures, build_type, is_staging, version_name):
+    def craft_assemble_release_task(self, architectures, build_type, is_staging, version_name, run_coverage=False):
```
Comment (mitchhentges) [inline automation/taskcluster/lib/tasks.py]: Don't we only want to do coverage checking on non-release builds?

File: automation/taskcluster/lib/tasks.py [hunk 3]
```diff
@@ -144,6 +158,9 @@ def craft_assemble_task(self, variant):
                 'symbol': 'A',
                 'tier': 1,
             },
+            scopes=[
+                "secrets:get:project/mobile/fenix/{}".format(variant.build_type)
```
Comment (mitchhentges) [inline automation/taskcluster/lib/tasks.py]: I don't think that separating by build type quite makes sense anymore.
Before, we used to just have secrets for each of the release channels (`production`/`beta`/`nightly`), in which case each of the release build types matched up one-to-one.
But, we are starting to see some PR-specific secrets, like this codecov secret and the nimbledroid secret.

I'm guessing that our secrets now fall into one of the following four buckets:
* `nightly`
* `beta`
* `production`
* `pr`

File: automation/taskcluster/upload_coverage_report.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# If a command fails then do not proceed and fail this script too.
+set -ex
+
+# Get token for uploading to codecov and append it to codecov.yml
+python automation/taskcluster/helper/get-secret.py \
+    -s project/mobile/fenix/codecov \
```
Comment (mitchhentges) [inline automation/taskcluster/upload_coverage_report.sh]: Sounds like this is in `project/mobile/fenix/pr` now?

File: automation/taskcluster/lib/tasks.py [hunk 5]
```diff
@@ -129,11 +129,22 @@ def craft_assemble_raptor_task(self, variant):
             },
         )
 
-    def craft_assemble_task(self, variant):
+    def craft_assemble_pr_task(self, variant, run_coverage=False):
+        coverage_flag = '-Pcoverage ' if run_coverage else ''
+        assemble_gradle_command = ('{} assemble{}'.format(coverage_flag, variant.for_gradle_command),)
+        post_gradle_command = ('automation/taskcluster/upload_coverage_report.sh' if run_coverage else '',)
```
Comment (mitchhentges) [inline automation/taskcluster/lib/tasks.py]: Understanding the order of operations here is a little tough, but I don't want to block your PR - I can take a look when I update the `nimbledroid` secret

File: automation/taskcluster/upload_coverage_report.sh (hunk 4 above)
Comment (colintheshots) [reply automation/taskcluster/upload_coverage_report.sh]: Fixed.

File: automation/taskcluster/lib/tasks.py (hunk 1 above)
Comment (colintheshots) [reply automation/taskcluster/lib/tasks.py]: Fixed.

File: automation/taskcluster/lib/tasks.py (hunk 2 above)
Comment (colintheshots) [reply automation/taskcluster/lib/tasks.py]: Fixed.

File: automation/taskcluster/lib/tasks.py (hunk 3 above)
Comment (colintheshots) [reply automation/taskcluster/lib/tasks.py]: Updated.

**Comment by mitchhentges** [comment]
I've landed the [permissions patch](https://phabricator.services.mozilla.com/D40893) and I'm just getting it applied atm

**Comment by mitchhentges** [comment]
The permissions patch is landed and applied, and you can verify that by seeing the [`fenix:pull-request` role](https://tools.taskcluster.net/auth/roles/repo%3Agithub.com%2Fmozilla-mobile%2Ffenix%3Apull-request)

**Pull request merged**